specs/Print.md

```diff
@@ -0,0 +1,816 @@
+  }
+  CHECK_FAILURE(printSettings2->put_ColorMode(printInfo.ColorMode));
+  CHECK_FAILURE(printSettings2->put_Collation(printInfo.Collation));
+  CHECK_FAILURE(printSettings2->put_Duplex(printInfo.Duplex));
+  CHECK_FAILURE(printSettings2->put_ScaleFactor(printInfo.ScaleFactor));
+  CHECK_FAILURE(printSettings2->put_ShouldPrintBackgrounds(printInfo.PrintBackgrounds));
+  CHECK_FAILURE(printSettings2->put_ShouldPrintHeaderAndFooter(printInfo.HeaderAndFooter));
+  CHECK_FAILURE(printSettings2->put_HeaderTitle(printInfo.HeaderTitle.c_str()));
+  CHECK_FAILURE(printSettings2->put_FooterUri(printInfo.FooterUri.c_str()));
+  CHECK_FAILURE(printSettings2->put_PrinterName(printerName.c_str()));
+
+  wil::unique_cotaskmem_string title;
+  CHECK_FAILURE(m_webView->get_DocumentTitle(&title));
+
+  CHECK_FAILURE(webView2_15->Print(
+      printSettings2.get(),
+      Callback<ICoreWebView2PrintCompletedHandler>(
+          [title = title.get(), this](HRESULT errorCode, BOOL isSuccessful) -> HRESULT
+          {
+            std::wstring message = L"";
+            if (errorCode == S_OK && isSuccessful)
+            {
+              message = L"Printing " + std::wstring(title) +
+                              L" document to printer is succedded";
+            }
+            else
+            {
+              if (errorCode == S_OK)
+              {
+                message = L"Selected printer is not found, not available, offline or error state.";
+              }
+              else if (errorCode == E_INVALIDARG)
+              {
+                message = L"Invalid settings provided for the specified printer";
+              }
+              else if (errorCode == E_ABORT)
+              {
+                message = L"Printing " + std::wstring(title) +
+                                  L" document already in progress";
+              }
+              else if (errorCode == E_FAIL)
+              {
+                message = L"Printing " + std::wstring(title) +
+                                  L" document to printer is failed";
+              }
+            }
+
+            AsyncMessageBox(message, L"Print to printer");
+
+            return S_OK;
+          })
+          .Get()));
+  return true;
+}
+```
+
+```c#
+async void PrintToPrinter()
+{
+  string printerName = GetPrinterName();
+  CoreWebView2PrintSettings printSettings = GetSelectedPrinterPrintSettings(printerName);
+  try
+  {
+    string title = webView.CoreWebView2.DocumentTitle;
+
+    bool isSuccessful = await webView.CoreWebView2.PrintAsync(printSettings);
+
+    if (isSuccessful)
+    {
+       MessageBox.Show(this, "Printing " + title + " document to printer is succeeded", "Print to printer");
+    }
+    else
+    {
+      MessageBox.Show(this, "Selected printer is not found, not available, offline or error state.", "Print to printer");
+    }
+  }
+  catch (NotImplementedException exception)
+  {
+    MessageBox.Show(this, "Printing is failed: " + exception.Message, "Print to printer");
```

Comment:
Again, the C# version doesn't handle the same errors as the C++ version. The two versions should be functionally equivalent.

Reply:
Please see following related comment below.